src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -333,6 +333,34 @@ public char PeekChar(int delta)
             return ch;
         }
 
+        public bool NextAre(string chars, int offset = 0)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(chars));
+            var n = chars.Length;
+            if (!CanGet(offset + n - 1)) return false;
+            for (var i = 0; i < n; i++)
+            {
+                if (chars[i] != PeekChar(offset + i)) return false;
+            }
+            return true;
+        }
+
+        public bool NextIs(char c, int offset = 0) => TryGet(out var ch, offset) && (ch == c);
+
+        public bool CanGet(int num = 0)
+        {
+            Debug.Assert(_offset + num >= 0);
+            //  Debug.Assert(num >= -MaxCharsLookBehind);
+            return _offset + num < _characterWindowCount;
+        }
+
+        public bool TryGet(out char ch, int num = 0)
+        {
+            var ok = CanGet(num);
+            ch = ok ? PeekChar(num) : default;
+            return ok;
+        }
```

Comment:
why do we need this with PeekChar being a thing

Reply:
This allows the refactoring of code that uses `CanGet( offset )` followed by a called to `PeekChar( offset)`.  Also C# uses an `InvalidCharacter` when out of range, VB doesn't.